src/librbd/Journal.cc

```diff
@@ -1349,6 +1350,10 @@ void Journal<I>::handle_replay_process_safe(ReplayEntry replay_entry, int r) {
   CephContext *cct = m_image_ctx.cct;
 
   m_lock.Lock();
+  assert(m_state == STATE_REPLAYING ||
+         m_state == STATE_FLUSHING_RESTART ||
+         m_state == STATE_FLUSHING_REPLAY);
```

Comment:
Looks like another one of those failures due to a fault injection in the messaging layer.  I don't think it's related to this change (I've been running it in a loop w/ fault injection and it hasn't repeated).